### ISSUE TITLE
Cognito: message templates do not accept custom placeholders

### ISSUE THREAD

**Issue by timreibe** [aws/aws-cdk#20092]
### Describe the bug

When creating a custom message template for the verification message (verification via link, not code), CDK throws an error saying the custom template must contain the template string ` '{##Verify Email##}'`. This behavior is not expected, since I can customize the link placeholder with any string in between the brackets and hashtags in the AWS Cognito Console.

```shell
jsii.errors.JavaScriptError: 
  Error: Verification email body must contain the template string '{##Verify Email##}'
      at UserPool.verificationMessageConfiguration (/private/var/folders/ln/3jl8p87x4m75zf067d38gbb40000gn/T/jsii-kernel-oug8Nn/node_modules/@aws-cdk/aws-cognito/lib/user-pool.js:379:23)
      at new UserPool (/private/var/folders/ln/3jl8p87x4m75zf067d38gbb40000gn/T/jsii-kernel-oug8Nn/node_modules/@aws-cdk/aws-cognito/lib/user-pool.js:220:50)
      at /private/var/folders/ln/3jl8p87x4m75zf067d38gbb40000gn/T/tmpya_gg9dy/lib/program.js:8412:58
      at Kernel._wrapSandboxCode (/private/var/folders/ln/3jl8p87x4m75zf067d38gbb40000gn/T/tmpya_gg9dy/lib/program.js:8840:24)
      at Kernel._create (/private/var/folders/ln/3jl8p87x4m75zf067d38gbb40000gn/T/tmpya_gg9dy/lib/program.js:8412:34)
      at Kernel.create (/private/var/folders/ln/3jl8p87x4m75zf067d38gbb40000gn/T/tmpya_gg9dy/lib/program.js:8153:29)
      at KernelHost.processRequest (/private/var/folders/ln/3jl8p87x4m75zf067d38gbb40000gn/T/tmpya_gg9dy/lib/program.js:9761:36)
      at KernelHost.run (/private/var/folders/ln/3jl8p87x4m75zf067d38gbb40000gn/T/tmpya_gg9dy/lib/program.js:9724:22)
      at Immediate._onImmediate (/private/var/folders/ln/3jl8p87x4m75zf067d38gbb40000gn/T/tmpya_gg9dy/lib/program.js:9725:46)
      at processImmediate (node:internal/timers:464:21)
```

### Expected Behavior

The email body shall accept any template string that is within `{##` and `##}` as a placeholder for the verification link.

### Current Behavior

CDK throws an error saying the email body **must** contain ` {##Verify Email##}`.

### Reproduction Steps

When setting up a Cognito UserPool in CDK for Python, set the following params:

```python
cognito.UserPool(
    scope=...,
    id=...,
    ...,
    user_verification=cognito.UserVerificationConfig(
        email_body="Hey du! {##Klicke hier##}, um deine E-Mail zu verifizieren und die Registrierung abzuschließen.",
        email_subject="Registrierungsbestätigung",
        email_style=cognito.VerificationEmailStyle.LINK
    )
)
```

### Possible Solution

https://github.com/aws/aws-cdk/blob/8da006ab551213ecbdb6dc26860fe90c1d2e95e2/packages/%40aws-cdk/aws-cognito/lib/user-pool.ts#L986

faulty condition:

```js
emailMessage.indexOf(VERIFY_EMAIL_TEMPLATE) < 0
```

Check if the placeholder template matches to a regex instead of searching for the `VERIFY_EMAIL_TEMPLATE` placeholder (`{##Verify Email##}`) in the string.

### CDK CLI Version

2.21.1

### Node.js Version

-

### OS

{"os":"linux","arch":"x64"}

### Language

Python

**Comment by corymhall** [comment]
@iamnotturner I agree, the solution here should be to use a regex to verify. Putting the message in
the UI here for reference

```
You can customize this message with HTML. "Verify email" is the text that will be displayed over the clickable link in the message. You can customize the "Verify email" string, but the variable - some text enclosed by "{##" and "##}" - must be kept in the message.
```

I am unassigning and marking this issue as `p2`, which means that we are unable to work on this immediately.

We use +1s to help prioritize our work, and are happy to revaluate this issue based on community feedback. You can reach out to the cdk.dev community on [Slack](https://join.slack.com/t/cdk-dev/shared_invite/zt-mso6p56d-qJp7SOTBvMaQuDrx7R2wHg) to solicit support for reprioritization.

**Comment by timwhunt** [comment]
+1 to please fix this.

**Comment by maxhayward** [comment]
+1

**Comment by ViniciusBastosTR** [comment]
+1

**Comment by pieterreuse** [comment]
+1

**Comment by awsluja** [comment]
+1

**Comment by Crusade1337** [comment]
+1

**Comment by jhones-dev** [comment]
Such small change to make, we just need a new regex logic. My native language is not even english, and I have to put "Verify Email" into it? In docs, it says that we can put anything if we respect the "{##" "##}", but in practice we can't change it.

**Comment by kdnakt** [comment]
Is this issue closed by this PR? https://github.com/aws/aws-cdk/pull/28832

**Comment by timreibe** [comment]
> Is this issue closed by this PR? #28832

 I haven't tried it yet, but yes, this should fix it. Thanks!
